application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -210,20 +202,19 @@ private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
 
         questionBuilder.append(originalQuestion);
 
-        StringBuilder tagBuilder = new StringBuilder();
-        int stringLength = questionBuilder.length();
-        for (ForumTag tag : threadChannel.getAppliedTags()) {
-            String tagName = tag.getName();
-            stringLength += tagName.length();
-            if (stringLength > MAX_QUESTION_LENGTH) {
-                break;
-            }
-            tagBuilder.append(String.format("%s ", tagName));
-        }
+        String tagsJoined = threadChannel.getAppliedTags()
+            .stream()
+            .map(ForumTag::getName)
+            .collect(Collectors.joining(" "))
+            .concat(" ");
 
-        questionBuilder.insert(0, tagBuilder);
+        String question = questionBuilder.insert(0, tagsJoined)
+            .toString()
+            .lines()
+            .limit(MAX_QUESTION_LENGTH)
+            .collect(Collectors.joining());
```

Comment:
Is this what you want, you changed the behaviour.

This limits it so there cant be more than 200 lines in the reply.
While the enhanced for checked or the ForumTag name's length was shorter than MAX_QUESTION_LENGTH

I'm not sure what's correct, but both don't make entirely sense to me, could you clarify what its meant to do?

Reply:
So this does change the behaviour slightly but I think the proposed stream missed what was going on. The length that was being checked in the old for loop was the length of the overall question, not the forum tags. We want questions to be under MAX_QUESTION_LENGTH since we don't want to burn through our ChatGPT credits.

This does change the behaviour slightly though. Ideally, I want the tags not to be added if the question exceeds the length limit. In this case, all tags are added and the length is shrunk to accommodate the tags. With this, we lose more of the context of the question (slowly by a few characters per tag). Is there a way to do this with Streams? Perhaps the old for loop may be better?

Reply:
```java
        for (ForumTag tag : threadChannel.getAppliedTags()) {
            String tagName = tag.getName();
            stringLength += tagName.length();
            if (stringLength > MAX_QUESTION_LENGTH) {
                break;
            }
            tagBuilder.append(String.format("%s ", tagName));
        }
```

The old for loop you only checked or the tag was longer than MAX_QUESTION_LENGTH

So the old for was also incorrect? I was confused about the check already either way.

If you wanna keep the stream, the filter has to change.
`tagBuilder + tagLength <=  MAX_QUESTION_LENGTH`


Since you'd want to check whenever the tag *can be added* to the tagBuilder right? Since the tagBuilder string can't be longer than MAX_QUESTION_LENGTH

Reply:
Sorry a little bit more context - if you look before the loop you see that I declare the stringLength variable to be the length of the questionBuilder (which includes the title and first message question). That is the stringLength I am checking against. So long as that original question plus the appended tag doesn't go over MAX_QUESTION_LENGTH then it should it would add it. (Sorry on mobile so it's hard to quote things)

Reply:
You're absolutely right, I missed the "+"

I'm on phone as well atm, writing the stream nicely with this would require me to think for a second.

I could take a look Tuesday.

Might end up getting returned to a for loop tbh, just from quick thinking I think a stream would be too cumbersome.